qt/scientific_interfaces/Indirect/test/IndirectFittingModelTest.h

```diff
@@ -457,29 +458,24 @@ class IndirectFittingModelTest : public CxxTest::TestSuite {
   void
   test_that_isPreviouslyFit_returns_true_if_the_spectrum_has_been_fitted_previously() {
     auto const model = getModelWithFitOutputData();
-    TS_ASSERT(model->isPreviouslyFit(0, 0));
+    TS_ASSERT(
+        model->isPreviouslyFit(TableDatasetIndex(0), IDA::WorkspaceIndex(0)));
   }
 
-  void test_that_hasZeroSpectra_returns_true_if_workspace_has_zero_spectra() {
+  void test_that_number_of_spectra_is_zero_if_workspace_has_zero_spectra() {
     auto model = getEmptyModel();
     auto const workspace = std::make_shared<Workspace2D>();
     SetUpADSWithWorkspace ads("WorkspaceEmpty", workspace);
 
-    model->addWorkspace("WorkspaceEmpty");
-
-    TS_ASSERT(model->hasZeroSpectra(0));
-  }
+    model->addWorkspace("WorkspaceEmpty", Spectra(""));
 
-  void
-  test_that_hasZeroSpectra_returns_true_if_the_dataIndex_provided_is_out_of_range() {
-    auto const model = createModelWithSingleWorkspace("WorkspaceName", 1);
-    TS_ASSERT(model->hasZeroSpectra(1));
+    TS_ASSERT(model->getSpectra(TableDatasetIndex(0)).size() == 0);
   }
 
   void
-  test_that_hasZeroSpectra_returns_false_if_workspace_contains_one_or_more_spectra() {
+  test_that_number_of_spectra_is_not_zero_if_workspace_contains_one_or_more_spectra() {
     auto const model = createModelWithSingleWorkspace("WorkspaceName", 1);
-    TS_ASSERT(!model->hasZeroSpectra(0));
+    TS_ASSERT(model->getSpectra(TableDatasetIndex(0)).size() != 0);
```

Comment:
is there a TS_ASSERT not equal functon??

Reply:
Yep it's called TS_ASSERT_DIFFERS and think it would be an improvement here.